src/tyr/route_serializer_osrm.cc

```diff
@@ -367,6 +417,35 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
         intersection->emplace("weight", json::fp_t{cost, 3});
     }
 
+    // Mark toll collection points
+    if (node->type() == TripLeg_Node_Type_kTollBooth) {
+      auto toll_collection_point_json = json::map({});
+      toll_collection_point_json->emplace("type", std::string("toll_booth"));
+      intersection->emplace("toll_collection", toll_collection_point_json);
+    }
+
+    // Mark border crossings
+    auto curr_country_code = etp->GetCountryCode(i);
+    auto prev_country_code = (i > 0) ? etp->GetCountryCode(i - 1) : "";
+    if (prev_country_code != "" && prev_country_code != curr_country_code) {
+      auto border_crossing_json = json::map({});
+      border_crossing_json->emplace("country_code", curr_country_code);
+      intersection->emplace("border_crossing", border_crossing_json);
```

Comment:
🤔 I'm wondering if we want to simply add the country_code and state_code wherever we have it instead of a `border_crossing` attribute. It would then be up to the client to infer a border crossing when one of the codes change.

Reply:
I pushed a commit with this change https://github.com/valhalla/valhalla/pull/2547/commits/f7c93f0f5813e26c922b25388449930da0e18573

Reply:
Looks like this change got reverted in a later commit 😞

Reply:
Removed in https://github.com/valhalla/valhalla/pull/2560